src/hook_manager.hh

```diff
@@ -125,6 +129,9 @@ public:
     CandidateList complete_hook_group(StringView prefix, ByteCount pos_in_token);
     void run_hook(Hook hook, StringView param, Context& context);
 
+    void add_linked(HookManager& hook_manager);
```

Comment:
I am considering whether these should be moved out into some `LinkableManager<T>` base class, but on the other hand it is not that much code duplication to give all five manager types the `(add|remove|m)_linked` members.

Reply:
Ha, I see that is exactly what you did with the `ScopeMember<T>` class in your implementation